pkg/build/controller/buildconfig/buildconfig_controller.go

```diff
@@ -136,7 +136,12 @@ func (c *BuildConfigController) handleBuildConfig(bc *buildapi.BuildConfig) erro
 		} else if buildgenerator.IsFatal(err) || kerrors.IsNotFound(err) || kerrors.IsBadRequest(err) || kerrors.IsForbidden(err) {
 			instantiateErr = fmt.Errorf("gave up on Build for BuildConfig %s due to fatal error: %v", bcDesc(bc), err)
 			utilruntime.HandleError(instantiateErr)
-			c.recorder.Event(bc, kapi.EventTypeWarning, "BuildConfigInstantiateFailed", instantiateErr.Error())
+			// Don't emit this event if the BuildRequest.LastVersion equals zero
+			// Caused by a race condition between the ImageChangeTrigger vs the BuildConfigChangeTrigger
+			// Fixes https://github.com/openshift/origin/issues/16557
+			if *request.LastVersion != int64(0) {
```

Comment:
request.LastVersion will always be zero in this codepath.  so you're going to mask any/all errors.

I think you need to actually look at the error you got from Instantiate (and/or make Instantiate throw a more distinctive error you can look for)